daml-lf/spec/daml-lf-1.rst

```diff
@@ -3094,41 +3094,93 @@ as described by the ledger model::
 
      'tpl' (x : T) ↦ { 'precondition' eₚ, … }  ∈  〚Ξ〛Mod
      eₚ[x ↦ vₜ]  ⇓  Err E
-   —————————————————————————————————————————————————————————————————————— EvUpdCreateErr1
+   —————————————————————————————————————————————————————————————————————— EvUpdCreateErr1a
      'create' @Mod:T vₜ ‖ S₀  ⇓ᵤ  (Err E, ε)
 
      'tpl' (x : T) ↦ { 'precondition' eₚ, … }  ∈  〚Ξ〛Mod
      eₚ[x ↦ vₜ]  ⇓  Ok 'False'
-   —————————————————————————————————————————————————————————————————————— EvUpdCreateFail
+   —————————————————————————————————————————————————————————————————————— EvUpdCreateFail1a
      'create' @Mod:T vₜ ‖ S₀
        ⇓ᵤ
      (Err (Fatal "Precondition failed on {Mod:T}."), ε)
 
-     'tpl' (x : T) ↦ { 'precondition' eₚ, 'agreement' eₐ, … }  ∈  〚Ξ〛Mod
+     'tpl' (x : T) ↦ { 'precondition' eₚ, …,
+        'implements' Mod₁:I₁ { … }, …, 'implements' Modₖ:Iₖ { … } }  ∈  〚Ξ〛Mod
      eₚ[x ↦ vₜ]  ⇓  Ok 'True'
+     'interface' (x₁ : I₁) ↦ { …, 'precondition' e₁ₚ, … } ∈ 〚Ξ〛Mod₁
+     e₁ₚ[x₁ ↦ 'to_interface' @Mod₁:I₁ @Mod:T vₜ]  ⇓  Ok 'True'
+     'interface' (x₂ : I₂) ↦ { …, 'precondition' e₂ₚ, … } ∈ 〚Ξ〛Mod₂
+     e₂ₚ[x₂ ↦ 'to_interface' @Mod₂:I₂ @Mod:T vₜ]  ⇓  Ok 'True'
+        …
+     'interface' (xᵢ : Iᵢ) ↦ { …, 'precondition' eᵢₚ, … } ∈ 〚Ξ〛Modᵢ
+     eᵢₚ[x₂ ↦ 'to_interface' @Modᵢ:Iᵢ @Mod:T vₜ]  ⇓  Err E
+   —————————————————————————————————————————————————————————————————————— EvUpdCreateErr1b
+     'create' @Mod:T vₜ ‖ S₀  ⇓ᵤ  (Err E, ε)
+
+     'tpl' (x : T) ↦ { 'precondition' eₚ, …,
+        'implements' Mod₁:I₁ { … }, …, 'implements' Modₖ:Iₖ { … } }  ∈  〚Ξ〛Mod
```

Comment:
```suggestion
        'implements' Mod₁:I₁ { … }, …, 'implements' Modₖ:Iₖ { … }, … }  ∈  〚Ξ〛Mod
```

Reply:
I like this idea but I changed the subscript to `i`, and left the other cases as `k`, to use that subscript only for the last  `implements` definition.